components/host-dashboard/AddFundsModal.js

```diff
@@ -359,7 +361,8 @@ const AddFundsModal = ({ collective, ...props }) => {
 
   // From the Collective page we pass host and collective as API v1 objects
   // From the Host dashboard we pass host and collective as API v2 objects
-  const canAddHostFee = host?.plan?.hostFees && collective.id !== host?.id;
+  const canAddHostFee =
+    host?.plan?.hostFees && collective.id !== host?.id && collective.parentCollective.id !== host?.legacyId;
```

Comment:
Looks weird that `collective.id` is v2 and `collective.parentCollective.id` is v1. What would be the explanation

Reply:
From the collective page we pass the `collective` as a v1 object. And `host` is always v2 as it's from the query within the `AddFundsModal`. Even the condition `collective.id !== host?.id ` is a bit weird in the sense in any collective page, this will be `false` as `host` is v2 and `collective` is v1. However that part is only relevant to the host dashboard where the comparison can be truthy. But since we check for `host?.plan?.hostFees` and that's always v2 we are good there. The comment above is misleading so I changed it a bit. 

My thinking is that we can simplify this condition to, `host?.plan?.hostFees && collective?.parentCollective?.id !== host?.legacyId` with no issues. I didn't do that here since that's not relevant to the issue at hand.

Reply:
> Even the condition collective.id !== host?.id is a bit weird in the sense in any collective page, this will be false as host is v2 and collective is v1. 

I don't know, you're the one who edited this condition the last and you should know better than anyone else. https://github.com/opencollective/opencollective-frontend/commit/8c00db6e560baf25a3129b939afae02aef21df38

We should assume that this condition is here for a good reason, please fix it and make it work whatever the case.

Reply:
I've checked all different pages (host dashboard and expense, transaction, project, even pages) involving both ICs as well as regular hosts. And found that there's no effect simplifying this condition to remove the `collective.id !== host?.id` part. I've made the change.

Reply:
Have you tried to Add Funds to a Fiscal Host?

Reply:
@znarf : Sorry for being late on this. This one too I tested and it seems that this change do not effect that. 

[adding-funds-to-host.webm](https://user-images.githubusercontent.com/12435965/228153035-a469bc26-7512-4bfc-b583-cc6bf5014d2e.webm)